### PR TITLE
#28 added test

### PR DESCRIPTION
**Pull request by apocarteres**
added test to prove H2Data does not support COLUMNS with ".", "-", or digits

### CODE REVIEW

**Comment by dmarkov** [comment]
Many thanks for the PR, let me find a reviewer for it

**Comment by dmarkov** [comment]
@krzyk this pull request is for you, please review

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 1]
```diff
@@ -134,7 +134,26 @@ public void createsTablesWithLongNames() throws Exception {
      */
     @Test
     public void supportsTableNamesWithIllegalCharacters() throws Exception {
-        new H2Data().with(".-.", new String[] {"pk"}, new String[0]);
+        new H2Data().with(".-.", new String[]{"pk"}, new String[0]);
+    }
+
+    /**
+     * H2Data supports column names with characters illegal to H2.
+     * todo: #28 user reports that H2Data does not support
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: todo should be moved after throws, and the format is `@tod #123:30min Do something`

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 2]
```diff
@@ -134,7 +134,26 @@ public void createsTablesWithLongNames() throws Exception {
...
+     * COLUMNS with ".", "-", or digits but i don't know for sure
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: todo should state what needs to be done, here if someone takes this todo he/she won't know what do do here.
You should ask issue creator for details.

**Comment by krzyk** [comment]
@apocarteres please see my comments above

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java (hunk 2 above)
Comment (apocarteres) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: issue creator expects <code>H2Data</code> supports such symbols. But i'm not really sure that it's correct. How to deal with that in this case?

Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: you could talk with issue creator to explain why it is needed, if you are still not convinced you could ask Yegor what he thinks

**Comment by apocarteres** [comment]
@krzyk Yegor answered in #28 that make a test is a good solution. I move the puzzle text to method body. Please check

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 3]
```diff
@@ -134,7 +134,28 @@ public void createsTablesWithLongNames() throws Exception {
...
+    }
+
+    /**
+     * H2Data supports column names with characters illegal to H2.
+     * @throws Exception In case test fails
+     */
+    @Test
+    @Ignore
+    public void supportsColumnNamesWithIllegalCharacters() throws Exception {
+        /* todo: #28:30min H2Data doesn't support COLUMNS with
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: todo should be in javadoc (so either at method, class or field level), move it just below the `@throws`.

Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: also, you can use longer lines, up to 80 characters per line, it reads better

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 4]
```diff
@@ -134,7 +134,28 @@ public void createsTablesWithLongNames() throws Exception {
...
+         * ".", "-" or digits
+         * but i don't know for sure
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: uppercase `i`

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 5]
```diff
@@ -134,7 +134,28 @@ public void createsTablesWithLongNames() throws Exception {
...
+         * should it support these symbols or not.
+         * It's needed to be confirmed and test should be uncommented
+         * when H2Data will be supporting mentioned symbols.
+         * */
+        final String key = "0-.col.-0";
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: inline key

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 6]
```diff
@@ -134,7 +134,28 @@ public void createsTablesWithLongNames() throws Exception {
...
+        final String table = "test";
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: inline table

**Comment by krzyk** [comment]
@apocarteres few comments above, and a question you are sure that H2Data shouldn't support such columns? If we introduced it for tables then doing the same for columns seams reasonable. If you agree please change the description of the todo puzzle, if not then leave it as it is.

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java (hunk 5 above)
Comment (apocarteres) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: can't do that
----------------
[ERROR] /src/test/java/com/jcabi/dynamo/mock/H2DataTest.java[152]: The String "0-.col.-0" appears 2 times in the file. (MultipleStringLiteralsCheck)

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java (hunk 6 above)
Comment (apocarteres) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: can't do that
----------------
[ERROR] /src/test/java/com/jcabi/dynamo/mock/H2DataTest.java[152]: The String "test" appears 2 times in the file. (MultipleStringLiteralsCheck)

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java (hunk 3 above)
Comment (apocarteres) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: can't out <code>todo</code> under <code>@throws</code>
it provokes check style error:
<pre>
[ERROR] /src/test/java/com/jcabi/dynamo/mock/H2DataTest.java[143]: Should contain one indentation space (MultilineJavadocTagsCheck)
</pre>

**Comment by apocarteres** [comment]
@krzyk please check last edits

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 7]
```diff
@@ -134,7 +134,25 @@ public void createsTablesWithLongNames() throws Exception {
...
+     * todo: #28:30min H2Data doesn't support COLUMNS with ".", "-" or digits
+     * but I don't know for sure should it support these symbols or not.
+     * It's needed to be confirmed and test should be uncommented
+     * when H2Data will be supporting mentioned symbols.
+     * @throws Exception In case test fails
+     */
+    @Test
+    @Ignore
+    public void supportsColumnNamesWithIllegalCharacters() throws Exception {
+        final String key = "0-.col.-0";
```
Comment (krzyk) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: I don't see where does this value exist in the file, have you inlined this variable and removed the `final String key`?
Same for the test table below, I don't see `"test"` string anywhere in the file, so if you inline it it should work (of course you have to remove the `table` variable)

**Comment by krzyk** [comment]
@apocarteres few comments above, and I still think that for the todo to be a good one you have to confirm that this functionality has to be added, right now the todo text is too vague. For example I would assume that this functionality has to be added (because it was accepted as a bug) so your todo should reflect that.

**Comment by apocarteres** [comment]
@krzyk 
1. checkstyle doesn't allow to inline 'test' and 'key' (complains about duplications)
2. I'm not even imaging what is <code>H2Data</code> how does it work and how should it work at all, therefore i can't say more in <code>todo</code>
3. All what i could make there i already made. Creating test to prove issue's create statement doesn't depend on some special knowledge of <code>H2Data</code>

**Comment by krzyk** [comment]
@apocarteres you are right regarding 1, I didn't notice that key and table is used twice in your test, as for 2 and 3, normally you should either: get that information from javadoc, or README.md, or from bug reported, if all of those sources fail you can create a bug about missing documentation about the class you need to change (see http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).
But creating a test is good enough.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmarkov** [comment]
@krzyk Many thanks! **16 mins** were added to your account in Transaction ID `51519623`; +16 added to your rating, at the moment it is: [+3842](http://www.netbout.com/b/35035?open=rating)

**Comment by dmarkov** [comment]
@rultor please deploy

**Pull request merged**